lms/djangoapps/commerce/utils.py

```diff
@@ -44,11 +44,24 @@ class EcommerceService(object):
     def __init__(self):
         self.config = CommerceConfiguration.current()
 
-    def is_enabled(self, user):
+    def is_enabled(self, user, allow_anonymous=False):
         """
-        Determines the availability of the Ecommerce service based on user activation and service configuration.
+        Determines the availability of the EcommerceService based on user activation and service configuration.
+
+        Keyword Arguments:
+            allow_anonymous: Boolean. Instructs is_enabled to allow anonymous users to interact with
+                             the EcommerceService, when indicated by the caller.
+
+        Returns:
+            Boolean
         """
-        return user.is_active and self.config.checkout_on_ecommerce_service
+        allow_user = user.is_active
+        anonymous_user = user.is_anonymous()
+        if allow_anonymous and anonymous_user:
+            allow_user = True
+
+        return allow_user and self.config.checkout_on_ecommerce_service
```

Comment:
Perhaps we can check with @mjfrey and see if there is a case where we would want to disable the EcommerceService specifically because the user is anonymous. If not, we could simplify the logic here.